rest/api_test.go

```diff
@@ -5023,3 +5023,72 @@ func TestImportOnWriteMigration(t *testing.T) {
 	response = rt.SendAdminRequest("PUT", "/db/doc1?rev=2-44ad6f128a2b1f75d0d0bb49b1fc0019", `{"value":"newer"}`)
 	assertStatus(t, response, http.StatusCreated)
 }
+
+func TestWebhookPropsWithAttachments(t *testing.T) {
+	wg := sync.WaitGroup{}
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		require.NoError(t, err, "Error reading request body")
+		require.NoError(t, r.Body.Close(), "Error closing request body")
+
+		var body db.Body
+		require.NoError(t, base.JSONUnmarshal(bodyBytes, &body), "Error parsing document body")
+		assert.Equal(t, "doc1", body[db.BodyId])
+		assert.Equal(t, "bar", body["foo"])
+
+		if strings.HasPrefix(body[db.BodyRev].(string), "1-") {
+			assert.Equal(t, "1-cd809becc169215072fd567eebd8b8de", body[db.BodyRev])
+		}
+
+		if strings.HasPrefix(body[db.BodyRev].(string), "2-") {
+			assert.Equal(t, "2-6433ff70e11791fcb7fdf16746f4b9e7", body[db.BodyRev])
+			attachments := body[db.BodyAttachments].(map[string]interface{})
+			attachment1 := attachments["attach1"].(map[string]interface{})
+			assert.Equal(t, "sha1-nq0xWBV2IEkkpY3ng+PEtFnCcVY=", attachment1["digest"])
+			assert.Equal(t, float64(30), attachment1["length"])
+			assert.Equal(t, float64(2), attachment1["revpos"])
+			assert.True(t, attachment1["stub"].(bool))
+			assert.Equal(t, "content/type", attachment1["content_type"])
+		}
+		wg.Done()
```

Comment:
Move this to the top of the handler with a defer, otherwise any failed `require.___` calls will prevent the waitgroup from releasing due to wg.Done() never being called.